benchmarks/benchmarks/ci/esmf_regridder.py

```diff
@@ -363,3 +364,96 @@ def time_lazy_regridding(self, cache):
     def time_regridding_realisation(self, cache):
         assert self.result.has_lazy_data()
         _ = self.result.data
+
+
+class TimeRegridderIO(MultiGridCompare):
+    def setup(self, type):
+        from esmf_regrid.experimental.io import load_regridder, save_regridder
+
+        self.load_regridder = load_regridder
+        self.save_regridder = save_regridder
+        from esmf_regrid.tests.unit.experimental.unstructured_scheme.test__mesh_to_MeshInfo import (
+            _gridlike_mesh,
+        )
```

Comment:
We implemented [`generate_data.py`](https://github.com/SciTools-incubator/iris-esmf-regrid/blob/91d640886766ac056892a56674d9d250bed5ed48/benchmarks/benchmarks/generate_data.py) to remove the need for lazy imports of test data generation, and to remove the risk that functions like `_gridlike_mesh` might change from commit to commit.

#124 hasn't been carried across to the `unstructured_scheme` branch yet. Short of a full branch update, perhaps you could cherry-pick the commit (fa2e34ba2a9ac2e6f306a19b469811a26bfad3b2) into this PR and implement a `_gridlike_mesh` generator in `generate_data.py`?

Since this approach necessitates saving/pickling objects, the synthetic data generation might be best used as part of `setup_cache()`. I expect such a pattern will gradually become more important anyway as we increase the volume of CI benchmarks and need them to run faster.

Reply:
Same goes for `_grid_cube` (although we got lucky and didn't need lazy imports in this case).

Reply:
I should add here that I think files and data generated by `save_regridder` should not follow this synthetic data generation pattern because we want to track the performance of save/load through changes in file format which we couldn't do if the data was fixed.

Reply:
It's hard to frame this in an easy rule of thumb! What we're running here is essentially a scientific experiment, where a change in result can either be:

- Interesting
  - Something to act on if it changes.
  - The thing you are benchmarking, which you therefore allow to change between commits.
  - e.g. the work that `save_regridder` does.
- Confounding
  - Something you are trying to see through to get to the Interesting stuff.
  - You want to eliminate their influence as much as possible, by *NOT* letting them change between commits.
  - e.g. benchmark source and target files/objects.